src/csharp/Microsoft.Spark/Services/ConfigurationService.cs

```diff
@@ -21,31 +21,45 @@ internal sealed class ConfigurationService : IConfigurationService
         public const string WorkerWriteBufferSizeEnvVarName =
             "spark.dotnet.worker.writeBufferSize";
 
+        internal const string WorkerVerDirEnvVarNameFormat = "DOTNET_WORKER_V{0}_DIR";
+
         private const string DotnetBackendPortEnvVarName = "DOTNETBACKEND_PORT";
         private const int DotnetBackendDebugPort = 5567;
 
         private const string DotnetNumBackendThreadsEnvVarName = "DOTNET_SPARK_NUM_BACKEND_THREADS";
         private const int DotnetNumBackendThreadsDefault = 10;
 
         private static readonly string s_procBaseFileName = "Microsoft.Spark.Worker";
-        private static readonly string s_procFileName =
-            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-            $"{s_procBaseFileName}.exe" :
-            s_procBaseFileName;
 
         private readonly ILoggerService _logger =
             LoggerServiceFactory.GetLogger(typeof(ConfigurationService));
 
         private string _workerPath;
 
+        /// <summary>
+        /// The Microsoft.Spark major assembly version is used to construct the
+        /// WorkerVerDirEnvVarNameFormat environment variable.
+        /// </summary>
+        internal static string WorkerVerDirEnvVarName { get; } =
+            string.Format(
+                WorkerVerDirEnvVarNameFormat,
```

Comment:
Just to be future proof, can we support all three levels? So we will try to go from patch version, minor version, then major version, then fall back to the defaut.

Reply:
Sure.   Which do you prefer?

```
DOTNET_WORKER_V1_0_0_DIR
DOTNET_WORKER_V1_0_DIR
DOTNET_WORKER_V1_DIR

DOTNET_WORKER_1_0_DIR
DOTNET_WORKER_1_0_0_DIR
DOTNET_WORKER_1_DIR
```

Reply:
Using the naming convention used in the lower half.

Reply:
Sgtm